src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/ExpressionLambdaRewriter.cs

```diff
@@ -303,7 +303,7 @@ private BoundExpression Indices(ImmutableArray<BoundExpression> expressions)
             foreach (var arg in expressions)
             {
                 var index = Visit(arg);
-                if (arg.Type != _int32Type)
+                if (index.Type != _int32Type)
                 {
                     index = ConvertIndex(index, arg.Type, _int32Type);
                 }
```

Comment:
Consider extracting the body of the loop to a `VisitIndex` method shared between the two uses.